tdxhydrorapid/inputs.py

```
@@ -130,17 +223,146 @@ def rapid_master_files(streams_gpq: str,
+                islands_dict[downstream].extend(island_ancestors)
+            else:
+                islands_dict[downstream] = list(island_ancestors)
+            # Set the downstream of the inlets
+            sgdf.loc[sgdf[id_field].isin(island_inlets), ds_id_field] = downstream
+        
+        # We need to check that no outlets are in other outlets
+        outlets = set(islands_dict.keys())
+        for outlet in outlets:
+            if outlet in to_delete:
+                # Find the real outlet
+                for key, items in islands_dict.items():
```

Ruff

Loop control variable `key` not used within loop body

(B007)